internal/talk/talk.go

```diff
@@ -0,0 +1,129 @@
+package talk
+
+// Talk compiles all the functionality related with the Talk product.
+// It is used by Corald to massage the data before sending to all other services it uses.
+
+import (
+	"errors"
+
+	"github.com/coralproject/shelf/internal/sponge/item"
+)
+
+var (
+	// ErrActionNotFound is action not found error.
+	ErrActionNotFound = errors.New("Action not found")
+
+	//ErrActionNotAllowed is an error when the action is not allowed.
+	ErrActionNotAllowed = errors.New("Action not allowed")
+
+	//ErrTypeNotExpected comes when the type asserted was not expected
+	ErrTypeNotExpected = errors.New("Type not expected")
+)
+
+//==============================================================================
+
+// AddAction an action (flag, like, etc) on an Item.
+func AddAction(spongedURL string, user item.Item, action string, targetID string) (item.Item, error) {
```

Comment:
Also, would be better to return a pointer to the item, and change any return due to errors with `nil, err`.